src/Symfony/Component/Serializer/Encoder/XmlEncoder.php

```diff
@@ -102,7 +121,10 @@ public function encode($data, string $format, array $context = [])
             $this->appendNode($this->dom, $data, $xmlRootNodeName);
         }
 
-        return $this->dom->saveXML($ignorePiNode ? $this->dom->documentElement : null);
+        $encodedString = $this->dom->saveXML($ignorePiNode ? $this->dom->documentElement : null);
+        $this->popDom();
```

Comment:
no need for new methods imho, this can be done inline

Reply:
@nicolas-grekas I can change popDom to inline, but should pushDom also inline?

Reply:
yes please